components/ai_chat/resources/page/components/uploaded_img_item/index.tsx

```diff
@@ -15,23 +17,30 @@ import * as Mojom from '../../../common/mojom'
 import styles from './style.module.scss'
 
 type Props = {
-  uploadedImage: Mojom.UploadedFile
+  uploadedImage?: Mojom.UploadedFile
   // removeImage is optional here so we can also reuse
   // this component in the conversation thread where remove
   // is not needed.
   removeImage?: () => void
+  isLoading?: boolean
```

Comment:
I'd prefer not to add mutually exclusive props if possible - this means we have to branch on pretty much everything we display in the component, and its not clear from the props what values are valid together.

Instead, might be worth having two separate components that both wrap a generic version of this.

Alternatively (and less preferred) we could strongly type the props as
```ts
type Props = {
    uploadedImage: Mojom.UploadedFile,
    removeImage?: () => void // if cancel isn't valid for uploads atm
  } | {
    isLoading: true
  }
```

conceptually though I think it'd make more sense to have an 

`<AttachmentSpinner/>` and `<AttachmentImage uploadedImage removeItem/>`

which both wrap an `<AttachmentItem title subtitle removeItem/>`

This fits how you use them too

Reply:
Done in https://github.com/brave/brave-core/pull/29017/commits/e4f9f89bd0bfd2fe06a2b46d1966d0b989a13d36

Reply:
Updated in https://github.com/brave/brave-core/pull/29017/commits/28670eb96e21b9ffdb37c31de54984b852c25bf9